pkg/commatrix/commatrix_test.go

```diff
@@ -0,0 +1,202 @@
+				},
+			},
+		},
+		AddressType: discoveryv1.AddressTypeIPv4,
+		Endpoints: []discoveryv1.Endpoint{
+			{
+				NodeName:  &testNode.Name,
+				Addresses: []string{"192.168.1.1", "192.168.1.2"},
+			},
+		},
+		Ports: []discoveryv1.EndpointPort{
+			{
+				Name:     nil,
+				Port:     func(i int32) *int32 { return &i }(80),
+				Protocol: func(p corev1.Protocol) *corev1.Protocol { return &p }(corev1.ProtocolTCP),
+			},
+		},
+	}
+)
+
+var _ = g.Describe("Commatrix Tests", func() {
+	g.Context("Create utils and client Matrix", func() {
+		g.BeforeEach(func() {
+			sch := runtime.NewScheme()
+
+			err := corev1.AddToScheme(sch)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = discoveryv1.AddToScheme(sch)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			fakeClient := fake.NewClientBuilder().WithScheme(sch).WithObjects(testNode, testPod, testService, testEndpointSlice).Build()
+			fakeClientset := fakek.NewSimpleClientset()
+
+			clientset = &client.ClientSet{
+				Client:          fakeClient,
+				CoreV1Interface: fakeClientset.CoreV1(),
+			}
+
+			ctrlTest = gomock.NewController(g.GinkgoT())
+			mockUtils = mock_utils.NewMockUtilsInterface(ctrlTest)
+			mockUtils.EXPECT().IsBMInfra().Return(false, nil)
+			mockUtils.EXPECT().IsSNOCluster().Return(false, nil)
+		})
+
+		g.AfterEach(func() {
+			ctrlTest.Finish()
+		})
+
+		g.It("should create with csv format ", func() {
+			var outputBuffer bytes.Buffer
+			originalStdout := os.Stdout
+			r, w, _ := os.Pipe()
+			os.Stdout = w
+
+			options := &CommatrixOptions{
+				format:              "csv",
+				customEntriesPath:   "",
+				customEntriesFormat: "",
+				debug:               false,
+				cs:                  clientset,
+				utilsHelpers:        mockUtils,
+			}
+			err := options.Run()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			w.Close()
+			os.Stdout = originalStdout
+			_, err = outputBuffer.ReadFrom(r)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			expectedHeader := "Direction,Protocol,Port,Namespace,Service,Pod,Container,Node Role,Optional\n"
+			actualOutput := outputBuffer.String()
+			o.Expect(actualOutput).To(o.HavePrefix(expectedHeader))
+
+		})
+
+		g.It("should create with json format", func() {
+			var outputBuffer bytes.Buffer
+			originalStdout := os.Stdout
+			r, w, _ := os.Pipe()
+			os.Stdout = w
+
+			options := &CommatrixOptions{
+				format:              "json",
+				customEntriesPath:   "",
+				customEntriesFormat: "",
+				debug:               false,
+				cs:                  clientset,
+				utilsHelpers:        mockUtils,
+			}
+
+			err := options.Run()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			w.Close()
+			os.Stdout = originalStdout
+			_, err = outputBuffer.ReadFrom(r)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			actualOutput := outputBuffer.String()
+
+			var jsonOutput interface{}
+			err = json.Unmarshal([]byte(actualOutput), &jsonOutput)
+			o.Expect(err).NotTo(o.HaveOccurred(), "output is not valid JSON")
+
+		})
+	})
+
+})
```

Comment:
What about validating the other cli functions?

Reply:
yah will do more